File: Python/Add Binary.py
class Solution:
    # @param a, a string
    # @param b, a string
    # @return a string
    def addBinary(self, a, b):
        a = self.convertBinToInt(a)
        b = self.convertBinToInt(b)
        print a,b
        return str(bin(a+b))[2:]
        
    def convertBinToInt(self, a):
        length = len(a)
        if length < 2:
            return int(a)
        ret = 0
        for i in range(length):
            if a[i] == "1":
                ret = ret + 2**i
        print ret
        return ret

s = Solution()
print s.addBinary("10100000100100110110010000010101111011011001101110111111111101000000101111001110001111100001101", "110101001011101110001111100110001010100001101011101010000011011011001011101111001100000011011110011")

File: Java/src/Add Binary.java
public class Solution {
    public String addBinary(String a, String b) {
        String ret = "";
        int plus = 0;
        for(int i = a.length()-1, j = b.length()-1 ; i>=0 || j>=0; i--,j--){
            int aInt = ((i<0)?0: Character.getNumericValue(a.charAt(i)));
            int bInt = ((j<0)?0: Character.getNumericValue(b.charAt(j)));
            ret = ((aInt ^ bInt)^plus) + ret;
            plus = ((aInt + bInt + plus) >1) ?1:0;
        }
        if (plus == 1)
            ret = '1'+ ret;
        return ret;
 
    }
}